code/modules/cargo/centcom_podlauncher.dm

```diff
@@ -788,18 +790,36 @@
 	QDEL_NULL(indicator)
 	. = ..()
 
-/datum/centcom_podlauncher/proc/supplypod_punish_log(var/list/whoDyin)
+/datum/centcom_podlauncher/proc/supplypod_punish_log(turf/target_turf, list/nearby_mobs, list/targeted_mobs, list/pod_contents)
 	var/podString = effectBurst ? "5 pods" : "a pod"
-	var/whomString = ""
-	if (LAZYLEN(whoDyin))
-		for (var/mob/living/M in whoDyin)
-			whomString += "[key_name(M)], "
-
-	var/msg = "launched [podString] towards [whomString]"
-	message_admins("[key_name_admin(usr)] [msg] in [ADMIN_VERBOSEJMP(specificTarget)].")
-	if (length(whoDyin))
-		for (var/mob/living/M in whoDyin)
+	var/whomstring_flw = ""
+	var/whomstring = ""
+	if (length(nearby_mobs))
+		var/list/keynames = list()
+		var/list/keynames_flw = list()
+		for (var/mob/living/M in nearby_mobs)
+			keynames += key_name(M)
+			keynames_flw += "[key_name_admin(M)][ADMIN_FLW(M)]"
+		whomstring = english_list(keynames, final_comma_text = ",")
+		whomstring_flw = english_list(keynames_flw, final_comma_text = ",")
+	var/hitstring = ""
+	var/mobs_differ = !compare_list(nearby_mobs, targeted_mobs)
+	if(length(targeted_mobs) && mobs_differ)
+		var/list/keynames = list()
+		for (var/mob/living/M in targeted_mobs)
+			keynames += key_name(M)
+		hitstring = " (at [english_list(keynames, final_comma_text = ",")])"
+
+	var/msg = "launched [podString] containing [english_list(pod_contents, final_comma_text = ",")]"
```

Comment:
Pod contents could be absolutely massive. The length of this should definitely be clamped.